library/src/main/java/com/datatorrent/lib/io/fs/HDFSFileSplitter.java

```diff
@@ -0,0 +1,184 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.io.fs;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.lib.io.block.BlockMetadata.FileBlockMetadata;
+
+/**
+ * HDFSFileSplitter extends {@link FileSplitterInput} to,
+ * 1. Add relative path to file metadata.
+ * 2. Ignore HDFS temp files (files with extensions _COPYING_).
+ * 3. Set sequencial read option on readers.
+ */
+public class HDFSFileSplitter extends FileSplitterInput
+{
+  private boolean sequencialFileRead;
+
+  public HDFSFileSplitter()
+  {
+    super();
+    super.setScanner(new HDFSScanner());
+  }
+
+  @Override
+  protected FileMetadata createFileMetadata(FileInfo fileInfo)
+  {
+    return new HDFSFileMetaData(fileInfo.getFilePath());
+  }
+
+  @Override
+  protected HDFSFileMetaData buildFileMetadata(FileInfo fileInfo) throws IOException
+  {
+    FileMetadata metadata = super.buildFileMetadata(fileInfo);
+    HDFSFileMetaData hdfsFileMetaData = (HDFSFileMetaData)metadata;
+
+    Path path = new Path(fileInfo.getFilePath());
+    FileStatus status = getFileStatus(path);
+    if (fileInfo.getDirectoryPath() == null) { // Direct filename is given as input.
+      hdfsFileMetaData.setRelativePath(status.getPath().getName());
+    } else {
+      String relativePath = getRelativePathWithFolderName(fileInfo);
+      hdfsFileMetaData.setRelativePath(relativePath);
+    }
+    return hdfsFileMetaData;
+  }
+
+  /*
+   * As folder name was given to input for copy, prefix folder name to the sub items to copy.
+   */
+  private String getRelativePathWithFolderName(FileInfo fileInfo)
+  {
+    String parentDir = new Path(fileInfo.getDirectoryPath()).getName();
+    return parentDir + File.separator + fileInfo.getRelativeFilePath();
+  }
+
+  @Override
+  protected FileBlockMetadata createBlockMetadata(FileMetadata fileMetadata)
+  {
+    FileBlockMetadata blockMetadta = new FileBlockMetadata(fileMetadata.getFilePath());
+    blockMetadta.setReadBlockInSequence(sequencialFileRead);
+    return blockMetadta;
+  }
+
+  public boolean isSequencialFileRead()
+  {
+    return sequencialFileRead;
+  }
+
+  public void setSequencialFileRead(boolean sequencialFileRead)
+  {
+    this.sequencialFileRead = sequencialFileRead;
+  }
+
+  /**
+   * HDFSScanner extends {@link TimeBasedDirectoryScanner} to ignore HDFS temporary files
+   * and files containing unsupported characters. 
+   */
+  public static class HDFSScanner extends TimeBasedDirectoryScanner
+  {
+    protected static final String HDFS_TEMP_FILE = ".*._COPYING_";
+    protected static final String UNSUPPORTED_CHARACTOR = ":";
+    private transient Pattern ignoreRegex;
```

Comment:
We already expose a property called "filePatternRegularExp" for patterns to accept. Having one more property for patterns to reject, could be confusing so didn't expose it.
Is provided reason good enough not to expose it as property or I should expose it?